app/org/sagebionetworks/bridge/validators/SurveySaveValidator.java

```diff
@@ -224,18 +263,30 @@ private void validateOneRuleSet(Errors errors, SurveyRule rule, Set<String> alre
         if (rule.getAssignDataGroup() != null) {
             actionCount++;
         }
-        if (actionCount != 1) {
-            errors.rejectValue(fieldPath, "must have one and only one action: skipTo, endSurvey, or assignDataGroup");
+        if (rule.getDisplayIf() != null) {
+            actionCount++;
         }
-        if (rule.getAssignDataGroup() != null && !dataGroups.contains(rule.getAssignDataGroup())) {
-            errors.rejectValue(fieldPath, "has a data group '" + rule.getAssignDataGroup()
-                    + "' that is not a valid data group: " + BridgeUtils.COMMA_SPACE_JOINER.join(dataGroups));            
+        if (rule.getDisplayUnless() != null) {
+            actionCount++;
         }
```

Comment:
My concern is that we don't know that this code is correctly counting displayIf and displayUnless unless we test it.